packages/auto-label/test/auto-label.ts

```diff
@@ -275,20 +275,164 @@ describe('auto-label', () => {
     });
   });
 
-  it('responds to backfill label event, backfilling issues with labels', async () => {
-    const payload = require(resolve(fixturesPath, './events/issue-labeled'));
-    const ghRequests = nock('https://api.github.com')
-      .get('/repos/testOwner/testRepo/issues')
-      .reply(200, [
-        {
+  describe('schedule repository', () => {
+    it('responds to a scheduled event', async () => {
+      const ghRequests = nock('https://api.github.com')
+        .log(console.log)
+        .get('/repos/testOwner/testRepo/issues')
+        .reply(200, {
           number: 1,
+        })
+        .get('/repos/testOwner/testRepo/issues/1/labels')
+        .reply(200)
+        .get('/repos/testOwner/testRepo/labels/myGitHubLabel')
+        .reply(200)
+        .post('/repos/testOwner/testRepo/labels', {
+          name: 'myGitHubLabel',
+          color: 'FEFEFA',
+        })
+        .reply(201, [
+          {
+            id: 1811802233,
+            node_id: 'MDU6TGFiZWwxODExODAyMjMz',
+            url:
+              'https://api.github.com/repos/sofisl/mergeOnGreenTest/labels/anotherLabel',
+            name: 'myGitHubLabel',
+            color: 'FEFEFA',
+            default: false,
+            description: null,
+          },
+        ])
+        .post('/repos/testOwner/testRepo/issues/1/labels')
+        .reply(200, [
+          {
+            id: 1811802233,
+            node_id: 'MDU6TGFiZWwxODExODAyMjMz',
+            url:
+              'https://api.github.com/repos/sofisl/mergeOnGreenTest/labels/anotherLabel',
+            name: 'myGitHubLabel',
+            color: 'FEFEFA',
+            default: false,
+            description: null,
+          },
+        ]);
+      handler.callStorage = async () => downloadedFile;
+      await probot.receive({
+        name: 'schedule.repository',
```

Comment:
What triggers these events? Also, does this mean every old issue will be automatically labeled? Seems like it would only need to be done once per repository, not nightly?

Reply:
Ah, good catch. So, in theory, this function was supposed to run nightly for every repo, for two reasons:

1. We wanted to institute a solution that eliminated the need for a 'backfill' label, and 
2. We wanted to make sure repos had the appropriate labels if their relationship to a product changed (i.e., if they covered a more specific product, we wanted the label to be updated and the old label to be removed).

But, in thinking through this a little more, I realized this would be wasteful and time consuming - both of these cases are fairly infrequent, to your point. So, I implemented a solution that writes the repository names to a file when it runs, and the function checks if the repo name already exists on that file. If it does, then the nightly function won't run for that repo. This way, we're only catching new repos or newly-named repos (i.e., in the case that we further specify a repo).

I'm not sure if this solution would work; take a look and let me know what you think!